packages/salesforce-adapter/src/filters/history_tracking.ts

```diff
@@ -0,0 +1,185 @@
+    Object.values(type.fields).find(async fieldDef => (await apiName(fieldDef)) === name)
+  )
+  const historyTrackingAddedChange = (fieldName: string): Change<Field> | undefined => {
+    const before = fieldByApiName(typeBefore, fieldName)?.clone()
+    const after = fieldByApiName(typeAfter, fieldName)?.clone()
+    if (after === undefined) {
+      log.error('The field %o was added to %o\'s history tracking, but does not exist', fieldName, typeAfter.elemID.getFullName())
+      return undefined
+    }
+    if (before === undefined) {
+      // The field was added, so it will be handled in the field's AdditionChange
+      return undefined
+    }
+
+    before.annotations[FIELD_HISTORY_TRACKING_ENABLED] = false
+    after.annotations[FIELD_HISTORY_TRACKING_ENABLED] = true
+    return toChange({ before, after })
+  }
+
+  const historyTrackingRemovedChange = (fieldName: string): Change<Field> | undefined => {
+    const before = fieldByApiName(typeBefore, fieldName)?.clone()
+    const after = fieldByApiName(typeAfter, fieldName)?.clone()
+    if (before !== undefined) {
+      log.warn('The field %o was removed from %o\'s history tracking, but did not exist', fieldName, typeAfter.elemID.getFullName())
+      before.annotations[FIELD_HISTORY_TRACKING_ENABLED] = true
+    }
+    if (after === undefined) {
+      // the field was removed from the object, no need to add a change beyond the RemovalChange that already exists
+      return undefined
+    }
+    after.annotations[FIELD_HISTORY_TRACKING_ENABLED] = false
+    return toChange({ before, after })
+  }
+
+  return [
+    ...addedTrackedFields
+      .filter(shouldProcessField)
+      .map(fieldName => historyTrackingAddedChange(fieldName))
+      .filter(valueUtils.isDefined),
+    ...removedTrackedFields
+      .filter(shouldProcessField)
+      .map(fieldName => historyTrackingRemovedChange(fieldName))
+      .filter(valueUtils.isDefined),
+  ]
+}
+
+/**
+ * Note: we assume this filter runs *after* custom objects are turned into types (custom_object_to_object_type) but
+ * *before* these types are split up into different elements (custom_type_split)
+ * */
+const filter: LocalFilterCreator = () => ({
+  onFetch: async elements => {
+    await awu(elements)
+      .filter(isObjectType)
+      .filter(isCustomObject)
+      .forEach(centralizeHistoryTrackingAnnotations)
+  },
+  preDeploy: async changes => {
+    const changedCustomObjectFields = changes
+      .filter(isAdditionOrModificationChange)
+      .map(getChangeData)
+      .filter(isField)
+      .filter(isFieldOfCustomObject)
+
+    const objectTypeChanges = changes
+      .filter(isAdditionOrModificationChange)
+      .filter(isObjectTypeChange)
+
+    // 1. For all CustomObjects, set the correct 'enableHistory' value
+    objectTypeChanges
+      .map(getChangeData)
+      .forEach(objType => {
+        objType.annotations[OBJECT_HISTORY_TRACKING_ENABLED] = isHistoryTrackingEnabled(objType)
```

Comment:
Isn't the data formatted as after the filter? Isn't this the correct version?
```suggestion
      .forEach(objType => {
        objType.annotations[OBJECT_HISTORY_TRACKING_ENABLED] = trackedFields(objType) !== undefined
```

Reply:
isHistoryTrackingEnabled works for both formats. I think using a named function makes the intention clearer.